# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../ipmiutil/getline.patch
# Copyright (C) 2009 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

*** ipmiutil-2.4.0/util/ipmicmd.c.orig	2009-08-06 16:39:00.000000000 +0200
--- ipmiutil-2.4.0/util/ipmicmd.c	2009-08-06 16:39:57.000000000 +0200
***************
*** 1116,1124 ****
  }
  #endif
  
! static char *getline(char *prompt, char fwipe)
  {
!    /* getline is the same format as readline, but much simpler, and portable. */
     static char linebuf[128];
     int c, i;
  
--- 1116,1124 ----
  }
  #endif
  
! static char *mygetline(char *prompt, char fwipe)
  {
!    /* mygetline is the same format as readline, but much simpler, and portable. */
     static char linebuf[128];
     int c, i;
  
***************
*** 1142,1150 ****
     return(linebuf);
  }
  
! static char *getline_wipe(char *prompt)
  {
!    return(getline(prompt,1));
  }
  
  void parse_lan_options(char c, char *optarg, char fdebugcmd)
--- 1142,1150 ----
     return(linebuf);
  }
  
! static char *mygetline_wipe(char *prompt)
  {
!    return(mygetline(prompt,1));
  }
  
  void parse_lan_options(char c, char *optarg, char fdebugcmd)
***************
*** 1198,1204 ****
                  }
                  break;
            case 'Y':      /* prompt for remote password */
! 		gpswd = getline_wipe("Enter IPMI LAN Password: ");
                  break;
            default:
                  break;
--- 1198,1204 ----
                  }
                  break;
            case 'Y':      /* prompt for remote password */
! 		gpswd = mygetline_wipe("Enter IPMI LAN Password: ");
                  break;
            default:
                  break;
